File: wisefy/src/androidTest/java/com/isupatches/wisefy/WiseFyTests.java
//package com.isupatches.wisefy;
//
//
//import com.google.android.gms.iid.InstanceID;
//import com.isupatches.wisefy.base.TestActivity;
//import org.junit.Test;
//import static org.junit.Assert.*;
//
//
//public class WiseFyTests extends BaseTestClass<TestActivity> {
//
//    public WiseFyTests() {
//        super(TestActivity.class);
//    }
//
//    @Test
//    public void builder_loggingFalse() {
//        WiseFy wiseFy = new WiseFy.brains(mActivityTestRule.getActivity()).logging(false).getSmarts();
//        assertEquals(false, wiseFy.mWiseFyConfiguration.isLoggingEnabled());
//        assertEquals(false, wiseFy.isLoggingEnabled());
//    }
//
//    @Test
//    public void builder_loggingTrue() {
//        WiseFy wiseFy = new WiseFy.brains(mActivityTestRule.getActivity()).logging(true).getSmarts();
//        assertEquals(true, wiseFy.mWiseFyConfiguration.isLoggingEnabled());
//        assertEquals(true, wiseFy.isLoggingEnabled());
//    }
//
//    @Test
//    public void calculateBars() {
//        int result = mWiseFy.calculateBars(-35, 5);
//        assertEquals(4, result);
//    }
//
//    @Test
//    public void compareSignalLevel() {
//        int result = mWiseFy.compareSignalLevel(-35, -70);
//        assertEquals(35, result);
//    }
//
//    @Test
//    public void olderGcm_IllegalAccessError_notThrown() {
//        InstanceID instanceID = InstanceID.getInstance(mActivityTestRule.getActivity());
//        assertNotNull(instanceID);
//    }
//}

File: wisefy/src/androidTest/java/com/isupatches/wisefy/WiseFyTests.kt
package com.isupatches.wisefy

import com.isupatches.wisefy.internal.base.BaseInstrumentationTest
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Test

/**
 * Tests that pertain to the builder for WiseFy or general use.
 *
 * @author Patches
 * @since 3.0
 */
internal class WiseFyTests : BaseInstrumentationTest() {

    companion object {
        private const val EXPECTED_NUMBER_OF_BARS = 4
        private const val EXPECTED_SIGNAL_RESULT = 35
    }

    @Test
    fun calculateBars() {
        val result = wisefy.calculateBars(TEST_RSSI_LEVEL_HIGH, TEST_NUMBER_OF_BARS)
        assertEquals(EXPECTED_NUMBER_OF_BARS.toLong(), result.toLong())
    }

    @Test
    fun compareSignalLevel() {
        val result = wisefy.compareSignalLevel(TEST_RSSI_LEVEL_HIGH, TEST_RSSI_LEVEL_LOW)
        assertEquals(EXPECTED_SIGNAL_RESULT.toLong(), result.toLong())
    }

    @Test
    fun getWiseFyLock() {
        assertNotNull(wisefy.getWiseFyLock())
    }
}
